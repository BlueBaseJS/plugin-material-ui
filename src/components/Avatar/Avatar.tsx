import { AvatarDefaultProps, AvatarProps, BlueBaseImage, Icon } from '@bluebase/components';

import MuiAvatar from '@material-ui/core/Avatar';
import React from 'react';
import { makeStyles } from '@material-ui/core';
import { useTheme } from '@bluebase/core';

const useStyles = makeStyles({
	root: ({ size: Size, textStyle, style }: AvatarProps) => {
		const size = Size || 56;
		return {
			...style as any,
			...textStyle,

			fontSize: size * 0.6,
			height: size,
			width: size,
		};
	},

	colorDefault: ({ color }: AvatarProps) => ({
		color,
	}),
});

export const Avatar: any = (props: AvatarProps) => {
	const { type, icon, size, image, text, ...rest } = props as any;

	const { theme } = useTheme();
	const classes = useStyles(props);
	const style: any = props.style;

	const color = props.color || theme.palette.background.default;

	if (type === 'icon') {
		return (
			<MuiAvatar classes={classes}  {...rest} style={style}>
				<Icon name={icon} color={color} size={size && size * 0.6} />
			</MuiAvatar>
		);
	}
	if (type === 'image') {
		return (
			<MuiAvatar classes={classes} {...rest} style={style}>
				<BlueBaseImage source={image} style={[{ height: size, width: size }, style]} />
			</MuiAvatar>
		);
	}

	return (
		<MuiAvatar  classes={classes} {...rest} style={{ ...style, color }}>
			{text}
		</MuiAvatar>
	);
};

Avatar.defaultProps = AvatarDefaultProps;
