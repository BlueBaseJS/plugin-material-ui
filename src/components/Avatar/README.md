# ü§† Avatar

Avatar control enables users to add images, icons or initials
within any application for instances such as a profile pictures.

## Usage

```jsx
<Avatar type="image" image={{ uri: 'https://picsum.photos/200' }} />
```

## Screenshots

Add screenshots here

## Supported Props

> Legend
    ‚úÖ = Fully Compatible and works as intended
    ‚ö†Ô∏è = Partially compatible
    ‚ùå = Not compatible

|  Prop | Compatible |           Comments           |
| :---: | :--------: | :--------------------------: |
|  type |     Ô∏è‚ö†Ô∏è    | Only works with `image` type |
|  size |     ‚ùå Ô∏è    |                              |
| color |     ‚ùå Ô∏è    |                              |
|  icon |     ‚ùå Ô∏è    |                              |
| image |     ‚ùå Ô∏è    |                              |
|  text |     ‚ùå Ô∏è    |                              |
| style |     ‚ùå Ô∏è    |                              |

## User Stories

|                              Story                              | In Storybook | Has Unit Test |
| :-------------------------------------------------------------: | :----------: | :-----------: |
|  User should be able to use Image Avatar by setting image path  |       ‚úÖ      |       ‚úÖ       |
| User should be able to use Image Avatar importing image in code |       ‚ùå      |       ‚ùå       |
