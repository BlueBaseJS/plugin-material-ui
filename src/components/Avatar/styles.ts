import { AvatarProps } from '@bluebase/components';
import { Theme } from '@bluebase/core';

export const styles = (
	{ color, size: Size, textStyle, style }: AvatarProps,
	_muiTheme: any,
	theme: Theme
) => {
	let finalColor = color;
	const size = Size || 56;

	if (
		color === 'primary' ||
		color === 'secondary' ||
		color === 'error' ||
		color === 'success' ||
		color === 'warning'
	) {
		finalColor = theme.palette[color].main;
	}

	return {
		root: {
			...style,
			...textStyle,

			fontSize: size * 0.6,
			height: size,
			width: size,
		},

		colorDefault: {
			color: finalColor,
		},
	};
};
