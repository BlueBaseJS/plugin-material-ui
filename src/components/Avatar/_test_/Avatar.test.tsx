import { Avatar } from '../Avatar';
import { BlueBaseApp } from '@bluebase/core';
import Plugin from '../../../index';
import React from 'react';
import { mount } from 'enzyme';
import { waitForElement } from 'enzyme-async-helpers';

const path = 'https://via.placeholder.com/300/09f.png%20C/O%20https://placeholder.com/';

test('avatar component should  return  source Image', async () => {
	const wrapper = mount(
		<BlueBaseApp plugins={[Plugin]}>
			<Avatar type="image" image={{ uri: path }} />
		</BlueBaseApp>
	);

	await waitForElement(wrapper, Avatar);

	const sourceProp: any = wrapper
		.find('Avatar BlueBaseImage')
		.first()
		.prop('source');
	expect(sourceProp.uri).toBe(path);
});

test('avatar component should  return  source Image', async () => {
	const wrapper = mount(
		<BlueBaseApp plugins={[Plugin]}>
			<Avatar type="image" image={path as any} />
		</BlueBaseApp>
	);

	await waitForElement(wrapper, Avatar);

	const sourceProp: any = wrapper
		.find('Avatar BlueBaseImage')
		.first()
		.prop('source');
	expect(sourceProp).toBe(path);
});

test('avatar component should  return   type =icon', async () => {
	const wrapper = mount(
		<BlueBaseApp plugins={[Plugin]}>
			<Avatar type="icon" icon="inbox" />
		</BlueBaseApp>
	);

	await waitForElement(wrapper, Avatar);

	const sourceProp: any = wrapper
		.find('Avatar Icon')
		.last()
		.prop('name');
	expect(sourceProp).toBe('inbox');
});

test('avatar component should  return  source Image', async () => {
	const wrapper = mount(
		<BlueBaseApp plugins={[Plugin]}>
			<Avatar type="text" text="N" />
		</BlueBaseApp>
	);

	await waitForElement(wrapper, Avatar);

	const sourceProp: any = wrapper
		.find('Avatar')
		.first()
		.text();
	expect(sourceProp).toBe('N');
});
