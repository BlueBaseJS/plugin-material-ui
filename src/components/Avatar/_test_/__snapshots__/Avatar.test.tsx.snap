// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar component should  return empty source 1`] = `
<WithStyles(Avatar)
  source=""
>
  <Text />
</WithStyles(Avatar)>
`;
